tap/tap.go

```diff
@@ -28,11 +28,25 @@ func Sync(ctx context.Context, logger kitlog.Logger, ol *OutputLogger, cl *clien
 
 		timeExtracted := time.Now().Format(time.RFC3339)
 		logger.Log("msg", "loading records", "time_extracted", timeExtracted)
+
 		records, err := stream.GetRecords(ctx, logger, cl)
 		if err != nil {
 			return err
 		}
 
+		// Get the enabled fields for this stream
+		disabledFields, err := catalog.GetDisabledFields(catalogEntry.Stream)
+		if err != nil {
+			return err
+		}
+
+		// Filter out the disabled fields from each record (ew)
```

Comment:
Rather than just doing this here - we should also maybe change the "Output" method on all streams to take a catalog and basically prune their own schema before we emit it.

Either that or add something to the output message that does similar?